src/wp-includes/script-loader.php

```diff
@@ -2372,20 +2372,8 @@ function wp_common_block_scripts_and_styles() {
 
 	wp_enqueue_style( 'wp-block-library' );
 
-	if ( current_theme_supports( 'wp-block-styles' ) ) {
-		if ( wp_should_load_separate_core_block_assets() ) {
-			$suffix = defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ? 'css' : 'min.css';
-			$files  = glob( __DIR__ . "/blocks/**/theme.$suffix" );
-			foreach ( $files as $path ) {
-				$block_name = basename( dirname( $path ) );
-				if ( is_rtl() && file_exists( __DIR__ . "/blocks/$block_name/theme-rtl.$suffix" ) ) {
-					$path = __DIR__ . "/blocks/$block_name/theme-rtl.$suffix";
-				}
-				wp_add_inline_style( "wp-block-{$block_name}", file_get_contents( $path ) );
```

Comment:
One thing that is not obvious from a quick review is how these styles are enqueued with the new approach. It'd be helpful for reviewers to provide some test instructions and explanation about these changes.

Reply:
@oandregal This style is output the block style is output. There is now a stylesheet called `"wp-block-{$block_name}-theme"` that is output when block is output, as this style is a dependency of the block. See `register_block_type_from_metadata`. 

I will add testing instructions.